### ISSUE TITLE
Customize ADB commands timeout

### ISSUE THREAD

**Issue by gaetanlgce** [ionic-team/native-run#259]
Hi,

On old devices, ADB invocation can be quite long to be executed, more than 5 seconds.
It could be useful to allow overriding the value of 5 seconds. (cf issue #5).

Currently, I edit the dist version of the file (https://github.com/ionic-team/native-run/blob/develop/src/android/utils/adb.ts#L469) in my node_modules to force another value of timeout.

Is there a proper way to pass a param for that. I don't use native-run directly but I use the ionic/cordova stack that use native-run. Maybe an environment variable ?

**Comment by me-to-you93** [comment]
Same problem here, would be interested by this option !

**Comment by markemer** [comment]
I'll look into this - seems like we could do this in a non-breaking way.

I'm not sure when I can get to it - but I'll put it on my radar. 

If you are interested in helping I'd be happy to review any PR you'd post.

**Comment by Dexus** [comment]
To bad that this never found its way to the code...

**Comment by piotr-cz** [comment]
The timeout argument would have to be read here:

https://github.com/ionic-team/native-run/blob/a6e56059ccd6f29ce60bc6de56012fd27eeeda5c/src/android/run.ts#L35

and passed all the way to here:

https://github.com/ionic-team/native-run/blob/a6e56059ccd6f29ce60bc6de56012fd27eeeda5c/src/android/utils/adb.ts#L358

**Comment by piotr-cz** [comment]
For me, it looks like that:

- the `startActivity` function executes ADB command which longer than 5s
- the `execAdb` function times out
- retrying fails, as the first ADB command is still executing
- app is deployed and started on device
- native-run exits with an error

```
npx capacitor run --live-reload --port 8100
...
√ Running Gradle build in 4.87s
× Deploying app-debug.apk to Pixel_9_API_35 - failed!
[error] Selected hardware device emulator-5554
        Installing xxx\android\app\build\outputs\apk\debug\app-debug.apk...
        Starting application activity xxx/xxx.MainActivity...
        ADBs is unresponsive after 5000ms, killing server and retrying...
        Error: Command failed: C:\Users\xxx\AppData\Local\Android\Sdk\platform-tools\adb -s emulator-5554 shell am
        start -W -n xxx/xxx.MainActivity
        adb.exe: device offline

        at ChildProcess.exithandler (node:child_process:422:12)
        at ChildProcess.emit (node:events:517:28)
        at maybeClose (node:internal/child_process:1098:16)
        at ChildProcess._handle.onexit (node:internal/child_process:303:5)
```

**Comment by piotr-cz** [comment]
I prepared PR #393 that increases timeout to 15_000 ms, because it works for my case.

@gaetanlgce @me-to-you93 @Dexus please let me know in above PR which timeout duration works for you

**Comment by Dexus** [comment]
I have patched it my own thanks to patch-package so I was able to set it to 30_000 that was working for me

**Comment by thoro23** [comment]
We have the same problem, the starting of the activity takes longer than 5s when deploying to an emulator. When deploying to a device it takes less than 5s and works fine.

For now we patch it with patch-package like @Dexus suggested but a proper fix would be preferable.

**Comment by xEverth** [comment]
Device offline and device still authorizing errors from ADB should be caught and retried on by native-run - increasing the timeout via environment variable or patch to startActivity just moves the issue slightly further without addressing it. 
Also native-run seems a little too trigger-happy with adb kill-server currently

**Comment by jordus100** [comment]
I and my team have the same issue, the timeout is too short on remote ADB connections. A nice solution would be to have the timeout as a parameter.
@xEverth How are retries supposed to fix anything if they'll hit the timeout each time?

**Comment by jordus100** [comment]
@xEverth I don't think you understand the full problem here. Some ADB commands simply take too long (for example on remote devices) and this is a different issue from simply handling the slowly changing ADB device state. For example if an `adb push` operation takes too long because of low bandwidth, simply retrying the operation will cause it to get stuck in a loop because it will hit the timeout each time. I think for cases like this, controlling the timeout with an env variable is entirely reasonable. It would enable each user to fine-tune the tradeoff between time and error sensitivity to their use case.
Honestly, hardcoding timeout values in production software interacting directly with a rich variety of differently set up hardware is an interesting choice, to put it mildly.
